src/define.zig

```diff
@@ -0,0 +1,159 @@
+const std = @import("std");
+
+const String = std.ArrayList(u8);
+const Database = std.StringHashMap(void);
+
+const State = struct {
+    allocator: std.mem.Allocator,
+    database: Database,
+    definitions: std.ArrayList(String),
+};
+
+pub fn define(
```

Comment:
Do you think it's worth adding some test coverage here?

Reply:
Done